### ISSUE TITLE
[Bug]: "Error sending scrobble to service" spam 

### ISSUE THREAD

**Issue by Nevinskas** [navidrome/navidrome#2831]
### I confirm that:

- [X] I have searched the existing [open AND closed issues](https://github.com/navidrome/navidrome/issues?q=is%3Aissue) to see if an issue already exists for the bug I've encountered
- [X] I'm using the latest version (your issue may have been fixed already)

### Version

0.51.0

### Current Behavior

The navidrome docker constantly spams this message (more than 100 per seconds):
```
time="2024-01-29T20:21:32Z" level=error msg="Error sending scrobble to service. Discarding" artist="Royal Blood" error=unrecoverable scrobbler=lastfm track="Little Monster" userId=e847b0f1-a43c-4748-be79-749200001f9f
time="2024-01-29T20:21:32Z" level=error msg="Error sending scrobble to service. Discarding" artist="Royal Blood" error=unrecoverable scrobbler=lastfm track="Little Monster" userId=e847b0f1-a43c-4748-be79-749200001f9f
time="2024-01-29T20:21:32Z" level=error msg="Error sending scrobble to service. Discarding" artist="Royal Blood" error=unrecoverable scrobbler=lastfm track="Little Monster" userId=e847b0f1-a43c-4748-be79-749200001f9f
```

and last fm scrobbling does not work.

### Expected Behavior

Not spamming messages and working scrobbling.

### Steps To Reproduce

I think this behavior happened after I upgraded navidrome

### Environment

```markdown
- OS: Arch + docker
- Browser: firefox
- Client: ultrasonic
```

### How Navidrome is installed?

Docker

### Configuration

```toml
navidrome:
    container_name: navidrome
    image: deluan/navidrome:latest
    restart: unless-stopped
    user: 1000:1000 # should be owner of volumes
    ports:
      - "4533:4533"
    environment:
      # Optional: put your config options customization here. Examples:
      ND_BASEURL: "/"
      ND_SCANSCHEDULE: 12h
      ND_LOGLEVEL: "info"
      ND_SESSIONTIMEOUT: 72h
      ND_LASTFM_ENABLED: "true"
      ND_LASTFM_APIKEY: "xxx"
      ND_LASTFM_SECRET: "xxx"
      ND_SPOTIFY_ID: "xxx"
      ND_SPOTIFY_SECRET: "xxx"
      #ND_ENABLETRANSCODINGCONFIG: "true"
      #ND_REVERSEPROXYWHITELIST: "0.0.0.0/0"
      #ND_BASEURL: ""
    volumes:
      - ./navidrome/data:/data
      - ./media/music:/music:ro
```

### Anything else?

I did try to delete the container itself and recreate it again, but that did not help.

### Code of Conduct

- [X] I agree to follow Navidrome's Code of Conduct

**Comment by metalheim** [comment]
Did you read the release notes of the latest navidrome release?
It states in there that you need to supply your own last.fm Apikey from version 0.51.0 and Re-enable scrobbling from navidromes user settings

**Comment by Nevinskas** [comment]
> Did you add your own Apikey after the latest navidrome release request you tu use your own?

No, I actually had it there since [v0.49.2](https://github.com/navidrome/navidrome/releases/tag/v0.49.2).

> It states in the release notes if version 0.51.0 that you need to Re-enable scrobbling from navidromes webui -> users

I've tried to disable via webui, relogin, restart docker. Did not help.
Also disabled via compose file (setting `ND_LASTFM_ENABLED` to false, restarted docker. Then the log messages disappeared. But after I enable it again, the log messages started to pour.

> Or maybe last.fm was just down in your region

Definitely not this - via phone with official last.fm app it scrobbling works.

 > enable log Level "debug"
 
``` 
 time="2024-01-30T09:41:18Z" level=debug msg="Loaded cache" dir=/data/cache/images numItems=472
time="2024-01-30T09:41:18Z" level=info msg="Finished initializing cache" cache=Image elapsedTime=2.12s maxSize=100MB
time="2024-01-30T09:41:19Z" level=debug msg="Sending scrobble" artist="Royal Blood" scrobbler=lastfm track="Little Monster"
time="2024-01-30T09:41:19Z" level=error msg="Error sending scrobble to service. Discarding" artist="Royal Blood" error="not authorized" scrobbler=lastfm track="Little Monster" userId=e847b0f1-a43c-4748-be79-749200001f9f
time="2024-01-30T09:41:19Z" level=debug msg="Sending scrobble" artist="Royal Blood" scrobbler=lastfm track="Little Monster"
time="2024-01-30T09:41:19Z" level=error msg="Error sending scrobble to service. Discarding" artist="Royal Blood" error="not authorized" scrobbler=lastfm track="Little Monster" userId=e847b0f1-a43c-4748-be79-749200001f9f
time="2024-01-30T09:41:19Z" level=debug msg="Sending scrobble" artist="Royal Blood" scrobbler=lastfm track="Little Monster"
time="2024-01-30T09:41:19Z" level=error msg="Error sending scrobble to service. Discarding" artist="Royal Blood" error="not authorized" scrobbler=lastfm track="Little Monster" userId=e847b0f1-a43c-4748-be79-749200001f9f
time="2024-01-30T09:41:19Z" level=debug msg="Sending scrobble" artist="Royal Blood" scrobbler=lastfm track="Little Monster"
time="2024-01-30T09:41:19Z" level=error msg="Error sending scrobble to service. Discarding" artist="Royal Blood" error="not authorized" scrobbler=lastfm track="Little Monster" userId=e847b0f1-a43c-4748-be79-749200001f9f
time="2024-01-30T09:41:19Z" level=debug msg="Sending scrobble" artist="Royal Blood" scrobbler=lastfm track="Little Monster"
time="2024-01-30T09:41:19Z" level=error msg="Error sending scrobble to service. Discarding" artist="Royal Blood" error="not authorized" scrobbler=lastfm track="Little Monster" userId=e847b0f1-a43c-4748-be79-749200001f9f
time="2024-01-30T09:41:19Z" level=debug msg="Sending scrobble" artist="Royal Blood" scrobbler=lastfm track="Little Monster"
```
Does not look that it helps.


I don't mind sharing personal navidrome files with you in private if you're interested to debug this problem.
Otherwise, I'll just delete all navidrome files ant start from scratch.

**Comment by deluan** [comment]
Two issues:
1) **Scrobbling not working**: Seems that your user is not authenticated with Last.fm, probably after you changed the API Key. For that you'll need to disable Scrobble in the Web UI, and re-enable it (no need to logoff, restart, nothing). When you re-enable it, you'll be redirected to Last.fm authorization page to connect your account with the API Key used by your instance of Navidrome.

2) **Log spam**: Not sure what is happening here, the unsuccessful scrobble should be discarded (as the message implies) and that's it. If this is still happening and you want to help troubleshoot this, are you willing to send a copy of your `navidrome.db.*` files to me? If yes, please send it to deluan @ navidrome.org and mention the issue in the subject. Thanks

**Comment by Nevinskas** [comment]
1. 
>For that you'll need to disable Scrobble in the Web UI, and re-enable it (no need to logoff, restart, nothing). When you re-enable it, you'll be redirected to Last.fm authorization page to connect your account with the API Key used by your instance of Navidrome.

Did that. Even when the scrobble is disabled via webui, the messages keep pouring. After authorization via last.fm does not change anything.

2.
>are you willing to send a copy of your navidrome.db.* files to me?

I don't mind that at all! Just did that. Also attached debug logs from the start.
Let me know if You need something else (i.e. core dump, try custom build etc.). Take Your time.

**Comment by deluan** [comment]
I think I'll need the logs from when the scrobble happen. Can you do the following steps:
1) Stop Navidrome and remove all pending scrobbles:
```
docker-compose stop navidrome 
sqlite3 navidrome.db
delete from scrobble_buffer;
```
2) Start Navidrome again 
3) Start playing one song.
4) Send me the logs from when you start to play the song till it starts the spam.

If you have questions, please let me know

**Comment by Nevinskas** [comment]
After deleting `scrobble_buffer`, I'm unable to reproduce this problem. Don't know how to reproduce it again.
If have any idea or need more info about something else -- let me know.
I think that this issue should be closed.
I'll keep my attention on scrobbling and if this issue happens again - I'll reopen the issue if that's okay.

Thank you for Your time.

**Comment by deluan** [comment]
Glad it worked. As I still have a copy of your DB, I sent you an email requesting some more info as a last attempt to reproduce it locally. If I can't, I think we are good to close this as it was probably some weird edge case in your environment.

**Comment by iglosiggio** [comment]
There's still an issue where track that are fail to be scrobbled stay on the buffer indefinitely

Example from my logs:
```log
time="2024-02-04T18:12:01-03:00" level=warning msg="Last.fm client.scrobble returned error" error="Post \"https://ws.audioscrobbler.com/2.0/?album=Summer%27s&albumArtist=Katsumi+Horii+Project&api_key=[REDACTED]&api_sig=a9ff8f18513e43fefe275eec002b5141&artist=Katsumi+Horii+Project&duration=295&format=json&mbid=&method=track.scrobble&sk=WCgSLlmuuoHfsyHDWDmt3zfeBsrvuluE&timestamp=1707081111&track=Green&trackNumber=2\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)" track=Green
time="2024-02-04T18:12:01-03:00" level=warning msg="Could not send scrobble. Will be retried" artist="Katsumi Horii Project" error="retry later" scrobbler=lastfm track=Green userId=953162b3-166d-4672-9a78-44b7e2721bcf
time="2024-02-04T18:12:06-03:00" level=debug msg="Sending scrobble" artist="Katsumi Horii Project" scrobbler=lastfm track=Green
time="2024-02-04T18:12:16-03:00" level=warning msg="Last.fm client.scrobble returned error" error="Post \"https://ws.audioscrobbler.com/2.0/?album=Summer%27s&albumArtist=Katsumi+Horii+Project&api_key=[REDACTED]&api_sig=a9ff8f18513e43fefe275eec002b5141&artist=Katsumi+Horii+Project&duration=295&format=json&mbid=&method=track.scrobble&sk=WCgSLlmuuoHfsyHDWDmt3zfeBsrvuluE&timestamp=1707081111&track=Green&trackNumber=2\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)" track=Green
time="2024-02-04T18:12:16-03:00" level=warning msg="Could not send scrobble. Will be retried" artist="Katsumi Horii Project" error="retry later" scrobbler=lastfm track=Green userId=953162b3-166d-4672-9a78-44b7e2721bcf
time="2024-02-04T18:12:21-03:00" level=debug msg="Sending scrobble" artist="Katsumi Horii Project" scrobbler=lastfm track=Green
time="2024-02-04T18:12:31-03:00" level=warning msg="Last.fm client.scrobble returned error" error="Post \"https://ws.audioscrobbler.com/2.0/?album=Summer%27s&albumArtist=Katsumi+Horii+Project&api_key=[REDACTED]&api_sig=a9ff8f18513e43fefe275eec002b5141&artist=Katsumi+Horii+Project&duration=295&format=json&mbid=&method=track.scrobble&sk=WCgSLlmuuoHfsyHDWDmt3zfeBsrvuluE&timestamp=1707081111&track=Green&trackNumber=2\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)" track=Green
time="2024-02-04T18:16:19-03:00" level=warning msg="Could not send scrobble. Will be retried" artist="Katsumi Horii Project" error="retry later" scrobbler=lastfm track=Green userId=953162b3-166d-4672-9a78-44b7e2721bcf
```

I think there should be a maximum amount of retries. This issue manifested today (probably because of some encoding issue from my files) and I ended up with like 10 scrobbles on the buffer waiting for a scrobble that kept failing half an hour ago.

**Comment by deluan** [comment]
@iglosiggio This is a complete different issue, and actually not an issue at all. Navidrome keeps retrying transient/temporary errors until it is successful, and automatically discards errors that are permanent (as per Last.fm docs). 

In your case above, seems that Navidrome was not able to connect to Last.fm, probably due to their API not being available (happens a lot) or a transient network issue. If ND was to discard after N retries, you could potentially lose those scrobbles.

If you still have any concerns, please open a new issue (or better yet, open a [discussion](https://github.com/navidrome/navidrome/discussions)), as it is unrelated to the this one.

**Comment by iglosiggio** [comment]
> @iglosiggio This is a complete different issue, and actually not an issue at all. Navidrome keeps retrying transient/temporary errors until it is successful, and automatically discards errors that are permanent (as per Last.fm docs).

Thanks for the response. As I said, I saw Navidrome retrying the same request for like half an hour. I didn't think it was a transient issue.

**Comment by tuhlaajapoika** [comment]
I'll jump in if anyone else comes to this thread with the same issue I had. I needed to reauthorize after I created an API key. I couldn't turn on LastFM scrobbling from Navidrome's WebUI because LastFM web auth returned with an error. Fixed it in my NGINX config: changed to `http2=off;` and LastFM's authorization returned successfully.

I had same `error="not authorized"` issue prior all that. To be on the safe side I removed the pending scrobbles from the database the way deluan directed. Everything's working now.
